spring-cloud-task-core/src/main/java/org/springframework/cloud/task/repository/support/DatabaseType.java

```diff
@@ -85,7 +86,12 @@ public enum DatabaseType {
 	/**
 	 * DB2AS400 DB.
 	 */
-	DB2AS400("DB2AS400");
+	DB2AS400("DB2AS400"),
+
+	/**
+	 * MariaDB DB.
+	 */
+	MARIADB("MariaDB");
```

Comment:
There is a DatabaseTypeTests unit test.   The tests need to be updated.

Reply:
I haven't tested the PR yet, But you will probably need to add the following file: spring-cloud-task-core/src/main/resources/org/springframework/cloud/task/schema-mariadb.sql   with the following DDL

```
 CREATE TABLE TASK_EXECUTION  (
	TASK_EXECUTION_ID BIGINT NOT NULL PRIMARY KEY ,
	START_TIME DATETIME DEFAULT NULL ,
	END_TIME DATETIME DEFAULT NULL ,
	TASK_NAME  VARCHAR(100) ,
	EXIT_CODE INTEGER ,
	EXIT_MESSAGE VARCHAR(2500) ,
	ERROR_MESSAGE VARCHAR(2500) ,
	LAST_UPDATED TIMESTAMP,
	EXTERNAL_EXECUTION_ID VARCHAR(255),
	PARENT_EXECUTION_ID BIGINT
) ENGINE=InnoDB;

CREATE TABLE TASK_EXECUTION_PARAMS  (
	TASK_EXECUTION_ID BIGINT NOT NULL ,
	TASK_PARAM VARCHAR(2500) ,
	constraint TASK_EXEC_PARAMS_FK foreign key (TASK_EXECUTION_ID)
	references TASK_EXECUTION(TASK_EXECUTION_ID)
) ENGINE=InnoDB;

CREATE TABLE TASK_TASK_BATCH (
 TASK_EXECUTION_ID BIGINT NOT NULL ,
 JOB_EXECUTION_ID BIGINT NOT NULL ,
	constraint TASK_EXEC_BATCH_FK foreign key (TASK_EXECUTION_ID)
	references TASK_EXECUTION(TASK_EXECUTION_ID)
)  ENGINE=InnoDB;

CREATE TABLE TASK_SEQ (
	ID BIGINT NOT NULL,
	UNIQUE_KEY CHAR(1) NOT NULL,
	constraint UNIQUE_KEY_UN unique (UNIQUE_KEY)
) ENGINE=InnoDB;

INSERT INTO TASK_SEQ (ID, UNIQUE_KEY) select * from (select 0 as ID, '0' as UNIQUE_KEY) as tmp;

CREATE TABLE TASK_LOCK  (
	LOCK_KEY CHAR(36) NOT NULL,
	REGION VARCHAR(100) NOT NULL,
	CLIENT_ID CHAR(36),
	CREATED_DATE DATETIME(6) NOT NULL,
	constraint LOCK_PK primary key (LOCK_KEY, REGION)
) ENGINE=InnoDB;
```

Reply:
[Also looking at the Spring framework code they are still treating mariadb as mysql.   i.e. 		jdbc/src/main/java/org/springframework/jdbc/support/JdbcUtils.java#L458-L461)